src/common/hist_util.cc

```diff
@@ -348,6 +370,28 @@ void DenseCuts::Init
   monitor_.Stop(__func__);
 }
 
+template<typename T>
+void GHistIndexMatrix::setIndexData(T* const index_data, size_t batch_threads, const SparsePage& batch, size_t rbegin, uint32_t* disps, size_t nbins)
```

Comment:
Do you do copy of the data here?

Reply:
no this template function is needed for  initial set of 'index' in GHistIndexMatrix